src/Pagination/PaginateDirective.php

```diff
@@ -89,7 +98,9 @@ public function manipulateFieldDefinition(DocumentAST &$documentAST, FieldDefini
     {
         $paginationManipulator = new PaginationManipulator($documentAST);
 
-        if ($this->directiveHasArgument('builder')) {
+        if ($this->directiveHasArgument('resolver')) {
+            $this->getResolverFromArgument('resolver');
+        } elseif ($this->directiveHasArgument('builder')) {
```

Comment:
This is fine, but let's also validate that only one of `model`, `builder` and `resolver` is used, else throw a `DefinitionException` indicating that they are mutually exclusive.

Reply:
Done but I'm not sure if you like the message and the fact it's always mentioning model and builder I could make it show it conditionally but didn't want to make the code dirty! 
Also added tests for it.

Reply:
I have had this issue a couple of time, please review my draft of how we can solve it once and for all: https://github.com/nuwave/lighthouse/pull/2233

Reply:
@spawnia Used your PR now, everything looks great, thanks! Please check the latest commit.